libs/blocks/article-feed/article-helpers.js

```diff
@@ -115,8 +114,7 @@ export function getTaxonomyModule() {
 }
```

Comment:
You may want to use a closure wrapped var that self initializes instead of a global.  This way you will not need to init `taxonomyModule` in article-header.js and tags.js.  It will require a change to `loadTaxonomy` to return `taxonomyModule` and a change in getLinkForTopic to `await` any calls to `getTaxonomyModule`:

```
export const getTaxonomyModule = (() => {
  let taxonomyModule;
  return async () => {
    if (!taxonomyModule) {
      taxonomyModule = await loadTaxonomy();
    }
    return taxonomyModule;
  };
})();
```

Reply:
I like this idea. But, this is going to affect a lot of methods and files which are using taxonomy methods. 
We need to change many methods to async functions and use Promises for loops. I wonder if we should make all the changes in this PR.

Reply:
It's fine to leave as is, but I'm only seeing it used in 2 places:
https://github.com/search?q=repo%3Aadobecom%2Fmilo%20getTaxonomyModule&type=code

Reply:
Yeah, but these changes are inside other methods which are inside other methods/modules, and so on.